jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard-actions.js

```diff
@@ -2,131 +2,137 @@
-
-        // disable scroll to bottom of notebook
-        scrollToBottom = IPython.Notebook.prototype.scroll_to_bottom;
-        IPython.Notebook.prototype.scroll_to_bottom = function() {};
+    function getFaIconClass(el) {
+        return $(el).attr('class').split(' ').filter(function(value) {
+            return /^fa-/.test(value);
+        })[0];
     }
 
-    function setViewButtonEnabled(state) {
-        var idx = VIEW_BTN_IDX[state];
-        $('#urth-dashboard-view-toolbar-buttons > button:nth-of-type(' + idx + ')')
-            .addClass('active')
-            .siblings().removeClass('active');
+    function updateAuthoringBtn(state) {
+        if (state !== STATE.NOTEBOOK &&
+            state !== STATE.DASHBOARD_PREVIEW) {
+            var $btns = $(toolbarBtnsSelector);
+            var icon = getFaIconClass($btns.find(
+                '.dashboard-layout-menu-item[data-dashboard-state="' + state + '"] i'));
+            $btns.find('.dashboard-authoring-btn')
+                .attr('data-dashboard-state', state)
+                .find('.dashboard-authoring-icon')
+                    .removeClass(function() {
+                        return getFaIconClass(this);
+                    })
+                    .addClass(icon);
+        }
     }
 
-    var isHeaderVisible = true;
-    var isToolbarVisible = true;
+    function setDashboardState(newState) {
+        if (newState !== currentState) {
+            if (newState === STATE.NOTEBOOK) {
+                exitDashboardMode();
+            } else {
+                enterDashboardMode(newState);
+            }
+            updateUIState(newState);
+            currentState = newState;
+        }
+    }
 
-    function toggleHeaders(doShow) {
+    function setHeaderVisibility(doShow) {
         // if hiding, save current state
         if (!doShow) {
             isHeaderVisible = $('#header-container').is(':visible');
             isToolbarVisible = $('div#maintoolbar').is(':visible');
         }
-        // hide or revert back to previous saved state
-        $('#header-container').toggle(doShow && isHeaderVisible);
-        $('.header-bar').toggle(doShow && isHeaderVisible);
+        // hide or revert back to previous state
+        $('#header-container, .header-bar').toggle(doShow && isHeaderVisible);
         $('div#maintoolbar').toggle(doShow && isToolbarVisible);
-        // make sure notebook resizes properly
         IPython.notebook.events.trigger('resize-header.Page');
     }
 
+    function setStateFromQueryString() {
+        // set 'Dashboard View' state if 'dashboard' query parameter is set
+        var idx = window.location.search.slice(1).split(/[&=]/).indexOf('dashboard');
+        if (idx !== -1) {
+            setDashboardState(STATE.DASHBOARD_PREVIEW);
+        }
+    }
+
+    function updateUIState(state) {
+        updateUrlState(state === STATE.DASHBOARD_PREVIEW);
+        setHeaderVisibility(state !== STATE.DASHBOARD_PREVIEW);
```

Comment:
Pull out these equality checks into their own variable? May make it easier to read this function and avoid doing same call multiple times:

    var inDashboardPreview = state === STATE.DASHBOARD_PREVIEW;